paasta_tools/marathon_tools.py

```diff
@@ -911,6 +916,14 @@ def _namespaced_get_classic_service_information_for_nerve(name, namespace, soa_d
     port_file = os.path.join(soa_dir, name, 'port')
     nerve_dict['port'] = service_configuration_lib.read_port(port_file)
     nerve_name = compose_job_id(name, namespace)
+    try:
+        with open(_get_classic_service_puppet_file(name)) as extras:
+            if os.fstat(extras.fileno()).st_size > 0:
+                nerve_dict.update(json.load(extras))
+    except IOError:
+        log.exception("Could not open {0} for reading".format(extras.name))
+    except ValueError:
+        log.exception("Could not interpret {0} as JSON".format(extras.name))
     return (nerve_name, nerve_dict)
 
 
```

Comment:
Is new line 933 going to actually allow you to achieve what you want here? As it forces a puppet/classic service to be announced under *every* smartstack namespace defined. (implying that you *must* define completely different services.

Is that ok with you?